modules/src/ics02_client/events.rs

```diff
@@ -71,8 +71,14 @@ pub fn extract_header_from_tx(event: &tendermint::abci::Event) -> Option<AnyHead
         let value = tag.value.as_ref();
         if let HEADER = key {
             let header_bytes = hex::decode(value).unwrap();
-            let header: AnyHeader = Protobuf::decode(header_bytes.as_ref()).unwrap();
-            return Some(header);
+            let result = match Protobuf::decode(header_bytes.as_ref()) {
```

Comment:
How about we just return Result directly and let the calling code in the relayer trace the underlying error?

Reply:
Maybe `try_from_tx` should see its signature changed from

```rust
pub fn try_from_tx(event: &tendermint::abci::Event) -> Option<IbcEvent> 
```

to

```rust
pub fn try_from_tx(event: &tendermint::abci::Event) -> (Option<IbcEvent>, Vec<Error>)
```
so that we can still build the `IbcEvent` even if the header decoding failed, but also return any errors encountered (eg. the header decoding error itself).